app/models/host/managed.rb

```diff
@@ -604,9 +604,7 @@ def hash_clone(value)
 
   def set_hostgroup_defaults
     return unless hostgroup
-    assign_hostgroup_attributes(%w{domain_id compute_profile_id})
```

Comment:
Does this prevent inheritance of the compute_profile_id in the case of an API request?  Should it still be inherited when possible, but we just skip the set_compute_attrs step?